gr-analog/python/analog/nbfm_tx.py

```diff
@@ -52,7 +52,7 @@ def __init__(self, audio_rate, quad_rate, tau=75e-6, max_dev=5e3, fh=-1.0):
         do_interp = audio_rate != quad_rate
 
         if do_interp:
-            interp_factor = quad_rate / audio_rate
+            interp_factor = int(quad_rate / audio_rate)           # force integer
```

Comment:
I'm not sure this is the correct solution to the problem, because if the quotient `quad_rate / audio_rate` is not an integer, then the block will appear to be working, but its output will be wrong because the interpolation factor will be wrong.

Still, something needs to be done, because in its present state this doesn't work at all, since `interp_factor` is always a `float` in Python3, so the interpolating FIR filter will complain.

Probably the problem dates back to GNU Radio 3.7 and Python 2.7, where the division of integers had the same semantics as the solution suggested by @duggabe here. But I guess it had the same problem I had mentioned above.

So I think we can take this as an opportunity for a better solution. Two options come to mind:
1. Check if the quotient is an integer and throw an error otherwise
2. Use a Polyphase arbitrary resampler to allow a non-integer `interp_factor`

I would prefer 1., because it is much simpler and doesn't change significantly the current code.

Reply:
If you look at lines 48 and 49, that check is already done, so it's just a matter of giving a warning or quitting.

I just checked with GRC 3.9.0.0-git trying to set a non-integral value and got
`Assertion "(quad_rate)%(audio_rate) == 0" failed.` 
so the user can not even execute the flowgraph with a bad value.

So all that's left is to do the type casting.

Reply:
My bad, sorry. Somehow I didn't see those lines before. Then I think your solution is correct.